UnityProject/Assets/Scripts/Systems/Research/Designs.cs

```diff
@@ -9,152 +9,170 @@
-	/// Stuff at spits out chemicals
-	/// </summary>
-	public Dictionary<string,int> ReagentsToMake { get; set; } //
+		/// <summary>
+		/// Stuff at spits out chemicals
+		/// </summary>
+		public Dictionary<string,int> ReagentsToMake { get; set; } //
 
-	/// <summary>
-	/// Self explanatory
-	/// </summary>
-	public String Description { get; set; } //
+		/// <summary>
+		/// Self explanatory
+		/// </summary>
+		public String Description { get; set; } //
 
-}
-
-public class Designs : MonoBehaviour
-{
-
-	void Awake()
-	{
-		if (!(Globals.IsInitialised))
-		{
-			new Task(JsonImportInitialization).Start();
-		}
 	}
-	public static class Globals
-	{
-		public static bool IsInitialised = false;
-		public static List<string> PathsList = new List<string> {
-			@"Metadata\Designs\AIDesigns",
-			@"Metadata\Designs\BiogeneratorDesigns",
-			@"Metadata\Designs\BluespaceDesigns",
-			@"Metadata\Designs\CompBoardDesigns",
-			@"Metadata\Designs\ComputerPartDesigns",
-			@"Metadata\Designs\ElectronicsDesigns",
-			@"Metadata\Designs\LimbgrowerDesigns",
-			@"Metadata\Designs\MachineDesigns",
-			@"Metadata\Designs\MechaDesigns",
-			@"Metadata\Designs\MechfabricatorDesigns",
-			@"Metadata\Designs\MedicalDesigns",
-			@"Metadata\Designs\MiningDesigns",
-			@"Metadata\Designs\MiscDesigns",
-			@"Metadata\Designs\PowerDesigns",
-			@"Metadata\Designs\StockPartsDesigns",
-			@"Metadata\Designs\TelecommsDesigns",
-			@"Metadata\Designs\WeaponDesigns",
-		};
-		public static Dictionary<string,Design> InternalIDSearch;
 
-	}
-	private static void JsonImportInitialization ()
+	public class Designs : MonoBehaviour
 	{
-		Globals.InternalIDSearch =  new Dictionary<string,Design>();
-		foreach (string FilePath in Globals.PathsList) {
-			string json = (Resources.Load (FilePath) as TextAsset).ToString ();
-			var JsonTechnologies = JsonConvert.DeserializeObject<List<Dictionary<String,System.Object>>> (json);
-			for (var i = 0; i < JsonTechnologies.Count (); i++) {
-				Design DesignPass = new Design ();
-				DesignPass.InternalID = JsonTechnologies [i] ["Internal_ID"].ToString ();
-				DesignPass.Name = JsonTechnologies [i] ["name"].ToString ();
-				DesignPass.ItemID = JsonTechnologies [i] ["Item_ID"].ToString ();
-
-				if (JsonTechnologies [i].ContainsKey ("Description")) {
-					DesignPass.Description = JsonTechnologies [i] ["Description"].ToString ();
-				} else {
-					DesignPass.Description = " *Breathes in through teeth* nahh, no description here mate ";
-				}
-
 
-				if (JsonTechnologies [i].ContainsKey ("Category")) {
-					DesignPass.Category = JsonConvert.DeserializeObject<List<string>> (JsonTechnologies [i] ["Category"].ToString ());
-				} else {
-					DesignPass.Category = new List<string> ();
-				}
+		void Awake()
+		{
+			if (!(Globals.IsInitialised))
+			{
+				JsonImportInitialization();
+			}
+		}
+		public static class Globals
+		{
+			public static bool IsInitialised = false;
+			public static List<string> PathsList = new List<string> {
```

Comment:
I'm not sure I like this hardcoded list of strings

Reply:
I've changed it to get the files inside the specified folder. The folder itself is hardcoded to Metadata/Designs. But we shouldn't be putting designs outside that folder as it is so I think its fine.